.github/workflows/run_tests.yml

```diff
@@ -15,6 +15,9 @@ jobs:
           - "3.8"
           - "3.9"
           - "3.10"
+          - "3.11"
+          - "3.12"
+          - "3.x"
```

Comment:
What does `3.x` mean, and why do you need it?

Reply:
`3.x` does not pass. Can you remove it?

Reply:
`3.x` means ["the most recent version of python"](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-the-python-version-input).  Right now, `3.x` is specifically `3.12.3`, which is the same as `3.12`.  So getting rid of `3.x` wouldn't solve the problem, because the `3.12` test would fail for the same reason.

I looked into the failure, though, and I'm pretty sure it was caused by a too-old version of `setuptools`. 
 Python 3.12 requires `setuptools>=66.1.0` (see pypa/setuptools#3685), but the version of `setuptools` used for CI was pinned to `65.6.3`.  Note that this is actually the same problem I tried to address in d6bbca6, but in that commit I only updated the `setuptools` version used to install `pymanopt` itself.  I didn't notice that there was another pin specifically for CI.  I just pushed a commit that updates this pin; hopefully that will work.